gapipy/resources/base.py

```diff
@@ -28,17 +28,17 @@ def __init__(self, data, stub=False, client=None):
     def options(cls, client):
         return APIRequestor(client, cls).options()
 
-    def fetch(self):
+    def fetch(self, httperrors_mapped_to_none=None):
         logger.info('Fetching %s/%s', self._resource_name, self.id)
 
         # Fetch the resource using the client bound on it, which handles cache get/set.
         resource_obj = getattr(self._client, self._resource_name).get(
             self.id,
             variation_id=getattr(self, 'variation_id', None),
-            httperrors_mapped_to_none=None)
-
-        self._fill_fields(resource_obj._raw_data)
-        self.is_stub = False
+            httperrors_mapped_to_none=httperrors_mapped_to_none)
+        if resource_obj:
+            self._fill_fields(resource_obj._raw_data)
+            self.is_stub = False
```

Comment:
🤔  This feels a bit weird because the `.get` method returns an instance of a Resource class; I'm wondering if instead of calling `self._fill_fields` it would be possible to just set the value of `self` to `resource_obj` ??? Is that possible in python?
```suggestion
        if resource_obj:
            self = resource_obj
```